org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/researchObject/ResearchObjectArchiveUploader.java

```diff
@@ -0,0 +1,258 @@
+
+                throw new Exception(responseMessage);
+            }
+        }
+    }
+
+    private static String getDatasetId(String response) throws Exception {
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode jsonResponse = mapper.readTree(response);
+
+        return jsonResponse.get("data").findValue("persistentId").textValue();
+    }
+
+    private static String getErrorMsg(String response) {
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode jsonResponse;
+        try {
+            jsonResponse = mapper.readTree(response);
+            String errorMsg = jsonResponse.get("message").textValue();
+            return errorMsg;
+        } catch (Exception e) {
+            return response;
+        }
+    }
+
+    private static String getJsonBody(ResearchObject ro, String privacyOption) throws IOException {
+
+        JsonNode template = getJsonTemplate();
+        ((ObjectNode) template.at("/datasetVersion/metadataBlocks/privacy/fields/0")).put("value", privacyOption);
+        JsonNode citationFields = template.at("/datasetVersion/metadataBlocks/citation/fields");
+        Iterator<JsonNode> fields = citationFields.elements();
+
+        while (fields.hasNext()) {
+            JsonNode field = fields.next();
+            switch (field.get("typeName").textValue()) {
+                case "title":
+                    ((ObjectNode) field).put("value", ro.getMetadata().getTitle());
+                    break;
+                case "author":
+                    ((ObjectNode) field.at("/value/0/authorName")).put("value", ro.getMetadata().getAuthor());
+                    break;
+                case "datasetContact":
+                    ((ObjectNode) field.at("/value/0/datasetContactEmail")).put("value", ro.getMetadata().getContact());
+                    break;
+                case "dsDescription":
+                    ((ObjectNode) field.at("/value/0/dsDescriptionValue")).put("value", ro.getMetadata().getDescription());
+                    break;
+                case "subject":
+                    ArrayNode arrayNode = ((ObjectNode) field).putArray("value");
+                    for (String subject : ro.getMetadata().getSubjects().getSubject()) {
+                        arrayNode.add(subject);
+                    }
+                    break;
+                case "publication":
+                    ResearchObject.Publication pub = ro.getPublication();
+                    if (pub != null) {
+                        Iterator<JsonNode> publicationFields = field.at("/value/0").elements();
+                        while (publicationFields.hasNext()) {
+                            JsonNode publicationField = publicationFields.next();
+                            String value;
+                            switch (publicationField.get("typeName").textValue()) {
+                                case "publicationCitation":
+                                    value = pub.getCitation();
```

Comment:
Please add curly bracer here and for the ifs

Reply:
Simplified the whole section